lib/bitvec.cpp

```diff
@@ -60,6 +61,7 @@ bitvec &bitvec::operator>>=(size_t count) {
 }
 
 bitvec &bitvec::operator<<=(size_t count) {
+    assert(static_cast<int>(count) >= 0);
```

Comment:
Yeah, I'd expect that `bad_alloc` would be thrown instead of `nullptr` being returned from `new`.

Of course, usually Linux will happily let you allocate a titanic amount of address space and just segfault or OOM-kill you when you actually touch the pages.